source/ui/memory-component-presenter.cpp

```diff
@@ -42,12 +42,33 @@ MemoryComponentPresenter::~MemoryComponentPresenter() {
 
 void MemoryComponentPresenter::onMemoryChanged(std::size_t address,
                                                std::size_t length) {
-  emit dataChanged(this->index(address / 1),
-                   this->index(address + length - 1));//  8bit
-  emit dataChanged(this->index(address / 2),
-                   this->index(address + length - 1));// 16bit
-  emit dataChanged(this->index(address / 4),
-                   this->index(address + length - 1));// 32bit
+  // calculate region for (max) 64bit memory cells
+  int start = address - (address % (64 / 8));
+  int end = start + (length - (length % (64 / 8)) + (64 / 8)) - 1;
+
+  // region should not exceed real memory size
+  if (end >= _memorySize) end = _memorySize - 1;
+  if (start <= 0) start = 0;
```

Comment:
`address` is `size_t` and `start` should therefore not be able to be negative (`n>n%m, n>=0`)